src/lib/ui/thermostat/ClimatePreset.tsx

```diff
@@ -0,0 +1,299 @@
+        typeof state.coolPoint === 'number'
+          ? toCelcius(state.coolPoint)
+          : undefined,
+      heating_set_point_celsius:
+        typeof state.heatPoint === 'number'
+          ? toCelcius(state.heatPoint)
+          : undefined,
+      hvac_mode_setting: state.hvacMode,
+    }
+
+    if (isCreate) {
+      createMutation.mutate(body)
+    } else {
+      updateMutation.mutate({
+        ...body,
+        manual_override_allowed: true,
+      })
+    }
+  }
+
+  return (
+    <div
+      {...attrs}
+      className={classNames('seam-thermostat-climate-preset', attrs.className)}
+    >
+      <ContentHeader title={preset.display_name} onBack={onBack} />
+
+      <div className='seam-main'>
+        <form
+          onSubmit={(e) => {
+            void handleSubmit(onSubmit)(e)
+          }}
+        >
+          {isCreate && (
+            <FormField>
+              <InputLabel>Key</InputLabel>
+              <TextField
+                size='large'
+                clearable
+                hasError={errors.key != null}
+                helperText={errors.key?.message}
+                inputProps={{
+                  ...register('key', {
+                    required: 'required',
+                    maxLength: {
+                      value: 20,
+                      message: 'max 20 chars',
+                    },
+                    minLength: {
+                      value: 3,
+                      message: 'min 3 chars',
+                    },
+                    validate(value) {
+                      const fixedValue = value.replace(/\s+/g, '').trim()
+                      return !otherPresets.some(
+                        (other) => other.climate_preset_key === fixedValue
+                      )
+                    },
+                  }),
+                }}
+              />
+            </FormField>
+          )}
+
+          <FormField>
+            <InputLabel>Display Name (Optional)</InputLabel>
+            <TextField
+              size='large'
+              clearable
+              hasError={errors.name != null}
+              helperText={errors.name?.message}
+              inputProps={{
+                ...register('name', {
+                  maxLength: {
+                    value: 20,
+                    message: 'max 20 chars',
+                  },
+                  minLength: {
+                    value: 3,
+                    message: 'min 3 chars',
+                  },
+                }),
+              }}
+            />
+          </FormField>
+
+          <FormField>
+            <InputLabel>Fan Mode</InputLabel>
+            <Controller
+              control={control}
+              name='fanMode'
+              render={({ field: { onChange, value } }) => (
+                <FanModeMenu
+                  block
+                  size='large'
+                  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                  mode={value!}
```

Comment:
This linter rule should not be bypassed. We need to understand why this can be null and put a proper fix in. Either `FanModeMenu` should accept nil or we should not try to render this. Same idea for the other cases.